lib/rexml/document.rb

```diff
@@ -443,6 +443,17 @@ def record_entity_expansion
       end
     end
 
+    attr_accessor :namespaces_cache
+
+    def enable_namespaces_cache
+      self.namespaces_cache = {}
+      begin
+        yield
+      ensure
+        self.namespaces_cache = nil
+      end
+    end
+
```

Comment:
Can we hide this API or make this API internal?

Reply:
OK
I see.